tenacity/__init__.py

```diff
@@ -406,10 +410,10 @@ def __iter__(self):
                 break
 
     @abstractmethod
-    def __call__(self, *args, **kwargs):
+    def __call__(self, fn: WrappedFn, *args: t.Any, **kwargs: t.Any) -> t.Any:
         pass
 
-    def call(self, *args, **kwargs):
+    def call(self, *args: t.Any, **kwargs: t.Any) -> t.Union[DoAttempt, DoSleep, t.Any]:
         """Use ``__call__`` instead because this method is deprecated."""
         warnings.warn(
```

Comment:
looks like reason to set deprecation period before remove (and also for bound function in Retrying)